### PR TITLE
Add ZIO#tapSome

### PR DESCRIPTION
**Pull request by kubukoz**

### CODE REVIEW

**Review by adamgfraser — DISMISSED** [review]
@kubukoz Looks good! Not sure why the rest of the CI checks aren't running. Can you try merging upstream changes in case there were any changes to CI settings that aren't incorporated in your version.

**Comment by kubukoz** [comment]
Rebased to latest master, let's see ;)

**Comment by kubukoz** [comment]
Apparently... doesn't work

**Review by adamgfraser — APPROVED** [review]
Thank you!

File: core/shared/src/main/scala/zio/ZIO.scala [hunk 1]
```diff
@@ -1814,6 +1814,19 @@ sealed trait ZIO[-R, +E, +A] extends Serializable with ZIOPlatformSpecific[R, E,
...
+   * If the partial function isn't defined at the input, the result is equivalent to the original effect.
+   *
+   * {{{
+   * readFile("data.json").tapSome {
+   *   case content if content.nonEmpty => putStrLn(content)
+   * }
+   * }}}
+   */
+  final def tapSome[R1 <: R, E1 >: E](f: PartialFunction[A, ZIO[R1, E1, Any]]): ZIO[R1, E1, A] =
+    self.tap(f.applyOrElse(_, (_: A) => ZIO.unit))
```
Comment (mijicd) [inline core/shared/src/main/scala/zio/ZIO.scala:1828]: Super nitpick, I'd try to use `ZIO.unitFn`.

**Comment by adamgfraser** [comment]
@kubukoz Thank you for your contribution and sorry for the delay on getting this merged!

**Pull request merged**